### PR TITLE
[POA] Handle the CollateralChecker not initialized exception

### PR DESCRIPTION
**Pull request by fassadlr**
This is an alternative to:

https://stratisplatformuk.visualstudio.com/StratisBitcoinFullNode/_workitems/edit/3782

It is better to work with a result class and then throw from the rule if the collateral checker is not yet fully initialized before trying to validate a block.

### CODE REVIEW

File: src/Stratis.Features.FederatedPeg/Collateral/CollateralChecker.cs [hunk 1]
```diff
@@ -22,7 +22,12 @@ public interface ICollateralChecker : IDisposable
         Task InitializeAsync();
 
         /// <summary>Checks if given federation member fulfills the collateral requirement.</summary>
-        bool CheckCollateral(IFederationMember federationMember);
+        /// <param name="federationMember">The federation member to check.</param>
+        /// <returns>
+        /// A result class indicating whether or not the collateral check was able to be done due to the it's initialization status
```
Comment (fassadlr) [inline src/Stratis.Features.FederatedPeg/Collateral/CollateralChecker.cs]:
```suggestion
        /// A result class indicating whether or not the collateral check was able to be done due to it's initialization status
```

**Comment by MithrilMan** [comment]
isn't better to rename Passed as Succeeded ?

**Comment by codingupastorm** [comment]
This is better, but if we receive a block whilst the CollateralChecker is starting up then we have a problem, we're going to deny incoming blocks. This could lead to peer banning?

We should:

A) Make sure we don't process received blocks until everything including the CollateralChecker is initialised. (Harder but good).

or

B) Wait in the rule for the CollateralChecker to finish initialising. (Easier but worse).

**Comment by fassadlr** [comment]
Closing as another implementation has been proposed.

**Pull request closed without merging**